taf/tests/test_updater/test_update/test_update_valid.py

```diff
@@ -400,3 +404,230 @@ def test_update_valid_when_no_upstream_when_contain_unsigned_commits(
         client_dir,
         no_upstream=True,
     )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_update_valid_dirty_index_auth_repo_update_file(origin_auth_repo, client_dir):
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    clients_auth_repo_path = client_dir / origin_auth_repo.name
+    update_file_without_commit(str(clients_auth_repo_path), "dirty_file.txt")
+
+    update_and_check_commit_shas(
+        OperationType.UPDATE,
+        origin_auth_repo,
+        client_dir,
+        force=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "origin_auth_repo",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+        },
+    ],
+    indirect=True,
+)
+def test_update_valid_dirty_index_auth_repo_add_file(origin_auth_repo, client_dir):
+    clone_repositories(
+        origin_auth_repo,
+        client_dir,
+    )
+
+    auth_repo_path = origin_auth_repo.path / "namespace/auth_repo"
+    add_file_without_commit(str(auth_repo_path), "new_file.txt")
```

Comment:
Make sure that you are modifying client's repo everywhere where we want to test if the update will be successful if client's repo is in some invalid state when called with `--force` flag